### PR TITLE
Token cancel commands

### PR DESCRIPTION
**Pull request by condron**
In process token cancelable commands.
Note: reviewing the code base cross process commands have been disabled since sometime in 2016, so no current feature loss here.

### CODE REVIEW

File: src/ReactiveDomain/ReactiveDomain.Tests/when_sending_commands.cs [hunk 1]
```diff
@@ -530,75 +532,57 @@ public void passing_commands_on_connected_busses_should_pass()
...
 
             Assert.Throws<CommandTimedOutException>(() =>
                  bus.Fire(new TestCommands.TestCommand(Guid.NewGuid(), null)));
             Assert.IsOrBecomesTrue(() => Interlocked.Read(ref gotCmd1) == 1, msg: "Expected Cmd1 handled");
             Assert.IsOrBecomesTrue(() => Interlocked.Read(ref cancelPublished) == 1, msg: "Expected cancel published");
-            Assert.IsOrBecomesTrue(() => Interlocked.Read(ref cancelRecieved) == 1, msg: "Expected cancel received");
         }
         [Fact]
-        public void tryfire_unsubscribed_commands_should_return_ack_timeout()
+        public void tryfire_unsubscribed_commands_should_return_throw_commandNotHandledException()
```
Comment (joshkempner) [inline src/ReactiveDomain/ReactiveDomain.Tests/when_sending_commands.cs]: "should_throw" not "should_return_throw"

File: src/ReactiveDomain/ReactiveDomain/Messaging/TokenCancellableCommand.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading;
+using ReactiveDomain.Bus;
+
+namespace ReactiveDomain.Messaging
+{
+    /// <summary>
+    /// A command that supports cancellation via cancelation tokens
+    /// n.b. this will only work inside a single process
```
Comment (joshkempner) [inline src/ReactiveDomain/ReactiveDomain/Messaging/TokenCancellableCommand.cs]: How can we support cancellable commands over the TCP/IP or http buses?

File: .gitignore [hunk 3]
```diff
@@ -253,3 +253,4 @@ paket-files/
 
 *.suo
 src/ReactiveDomain/.vs/ReactiveDomain/v14/.suo
+/src/ReactiveDomain/ReactiveDomain/Fakes/ReactiveUI.fakes
```
Comment (joshkempner) [inline .gitignore]: Your csproj file includes this, but you have it excluded from Git. Something's not right here.

File: src/ReactiveDomain/ReactiveDomain/ReactiveDomain.csproj [hunk 4]
```diff
@@ -85,6 +88,9 @@
       <HintPath>..\..\..\packages\reactiveui-core.6.5.2\lib\Net45\ReactiveUI.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="ReactiveUI.6.5.0.0.Fakes">
```
Comment (joshkempner) [inline src/ReactiveDomain/ReactiveDomain/ReactiveDomain.csproj]: Should this be added to the NuGet packages file?

File: src/ReactiveDomain/ReactiveDomain/Messaging/TokenCancellableCommand.cs (hunk 2 above)
Comment (condron) [inline src/ReactiveDomain/ReactiveDomain/Messaging/TokenCancellableCommand.cs]: We would have a cancel msg sent by the bus that would then trigger a cancelation source on the far side. I just walk through some patterns on this with Yves.

File: src/ReactiveDomain/ReactiveDomain/ReactiveDomain.csproj (hunk 4 above)
Comment (condron) [inline src/ReactiveDomain/ReactiveDomain/ReactiveDomain.csproj]: I'll take a look

Comment (condron) [reply src/ReactiveDomain/ReactiveDomain/ReactiveDomain.csproj]: not needed

File: .gitignore (hunk 3 above)
Comment (condron) [reply .gitignore]: not needed, I'll remove it

File: src/ReactiveDomain/ReactiveDomain/Messaging/TokenCancellableCommand.cs (hunk 2 above)
Comment (joshkempner) [reply src/ReactiveDomain/ReactiveDomain/Messaging/TokenCancellableCommand.cs]: OK

File: src/ReactiveDomain/ReactiveDomain.Tests/can_cancel_commands_via_cancellation_token.cs [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+            TestQueue.WaitFor<TestTokenCancellableCmd>(TimeSpan.FromSeconds(2));
+            TokenSource.Cancel();
+            TestQueue.WaitFor<Canceled>(TimeSpan.FromSeconds(2));
+            TestQueue.Commands
+                        .AssertNext<TestTokenCancellableCmd>(_ => true)
+                        .AssertNext<NestedTestTokenCancellableCmd>(_ => true)
+                        .AssertEmpty();
+            TestQueue.Responses
+                        .AssertNext<Canceled>(_ => true)
+                        .AssertNext<Fail>(_ => true)
```
[the remark this answers is not in the document]
Comment (joshkempner) [reply src/ReactiveDomain/ReactiveDomain.Tests/can_cancel_commands_via_cancellation_token.cs]: This is resolved by other tests that verify this behavior is correct.

File: src/ReactiveDomain/ReactiveDomain.Tests/can_cancel_commands_via_cancellation_token.cs [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+            Thread.Sleep(500);
+            TokenSource.Cancel();
+            TestQueue.WaitFor<Canceled>(TimeSpan.FromSeconds(2));
+            TestQueue.Commands
+                        .AssertNext<TestTokenCancellableCmd>(_ => true)
+                        .AssertNext<NestedTestTokenCancellableCmd>(_ => true)
+                        .AssertEmpty();
+            TestQueue.Responses
+                        .AssertNext<Canceled>(_ => true)
+                        .AssertNext<Fail>(_ => true)
```
[the remark this answers is not in the document]
Comment (joshkempner) [reply src/ReactiveDomain/ReactiveDomain.Tests/can_cancel_commands_via_cancellation_token.cs]: This is resolved by other tests that verify this behavior is correct.

**Pull request merged**